feat: add support for Streamz content provider

pejeio opened the pull request:
First off, I’d like to thank you for the excellent work on this project! This PR adds support for Streamz. I tested it end-to-end by:

- Pressing the download button on the Streamz website and confirming it entered the queue.
- Verifying that the downloader picked up the request from the queue.
- Successfully downloading 5 different series to confirm everything works as expected.

The logic for Streamz is quite similar to VTMGO, so I’ve followed the same approach for the most part. The main difference is in the way I handle submitting the login form. In the case of VTMGO,[ the submit button is pressed after entering the email](https://github.com/BelgianNoise/dl-utils/blob/49075f03f764490890067a2fb8e12985cf42cb8a/dl-downer/src/downloaders/VTMGO.py#L54-L55), while in the Streamz implementation, I wait until both the email and password are entered before triggering the submit. Maybe there’s a particular edge case or behavior that necessitates this? If there’s a good reason for it, I’m more than happy to update the Streamz implementation to match that logic.

BelgianNoise commented:
Hey, thanks for opening a PR. Very unexpected to see someone else in here :)

At a first glance all looks fine, but I will take a closer look over the weekend, and also test it out.
Would you mind sharing your streamz login with me just so I can test it out ? you can find my email on my gh profile. (if not, thats totally fine)

As a first comment I would like you to take a look at overlapping code from VTMGO and STREAMZ and possibly extract some of that logic to a function.
I don't mind having the streamz file call a function in the VTMGO file as they are almost the same thing.

---

With VTMGO you are required to press the submit button before entering the password, as their form will show different inputs based on wether you have an account or not.

pejeio commented:
Hi @BelgianNoise,

Thanks for the feedback! I've updated the code as requested and extracted the shared logic between VTMGO and Streamz into reusable functions. The refactoring moves all the common DRM handling, manifest parsing, filename generation, and subtitle processing logic to shared functions in `VTMGO.py`, which `STREAMZ.py` now imports and uses.

Regarding the Streamz login, I'd prefer not to share my credentials as they contain personal information. However, you can get a Streamz Basic subscription for just €8.99/month to test it out yourself if needed. The basic plan should be sufficient for testing the download functionality.

BelgianNoise commented on dl-downer/src/downloaders/STREAMZ.py:
```diff
@@ -0,0 +1,213 @@
+            submitButton = page.wait_for_selector('button:has-text("Log in")')
+            submitButton.click()
+
+            page.wait_for_selector("li.nav__item--userdropdown", timeout=200000000)
```
/

BelgianNoise commented on dl-downer/src/downloaders/STREAMZ.py:
```diff
@@ -0,0 +1,108 @@
+            assert os.getenv("AUTH_STREAMZ_PASSWORD"), "AUTH_STREAMZ_PASSWORD not set"
+            passwordInput.type(os.getenv("AUTH_STREAMZ_PASSWORD"))
+
+            submitButton = page.wait_for_selector('button:has-text("Log in")')
```
not a huge fan of this selector, but i dont see another straight forward way.

BelgianNoise commented on dl-downer/src/downloaders/STREAMZ.py:
```diff
@@ -0,0 +1,108 @@
+import os
```
tab size = 2 spaces

BelgianNoise commented:
@pejeio Big thanks for adding this to the repo. It's great to see someone contributing rather then taking the code and expanding on it in a private repo like some have done before.

This was on my to-do list for the summer, and now I don't have to :)

Is there any way I can contact you on another platform ? (Telegram, Signal, discord, ...)
I would love to talk more and stay in touch. I really think we could help each other out in the future.
